include/seqan3/io/sam_file/format_sam_base.hpp

```diff
@@ -459,16 +459,34 @@ inline void format_sam_base::read_header(stream_view_type && stream_view,
                                          ref_seqs_type & /*ref_id_to_pos_map*/)
 {
     auto it = std::ranges::begin(stream_view);
+    auto end = std::ranges::end(stream_view);
+    std::string string_buffer{};
```

Comment:
Please use vector here, because std::string can be quite slower because of the small string optimization